src/core/runtime.cpp

```diff
@@ -1016,10 +1024,12 @@ void validateConnection(ValidationContext &ctx) {
         shards::arrayPush(data.shared, type);
       }
     }
-    // and inherited
-    for (auto &info : ctx.inherited) {
-      for (auto &type : info.second) {
-        shards::arrayPush(data.shared, type);
+    if (!ctx.wire->pure) {
```

Comment:
Chan you change this to `if (!ctx.wire || !ctx.wire->pure) {` so data.wire can be null when composing?

I have a case like this in the shader translator where I take a vector of shards and have to compose them but they don't belong to a wire so it's set to null in SHInstanceData (also why it crashes on CI)